src/arrayMethodJoin.js

```diff
@@ -4,9 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      switch (true) {
+        case ((i === (this.length) - 1) && this[i] === undefined):
+          str += '';
+          break;
+        case (i === (this.length) - 1):
+          str += this[i];
+          break;
+        case (this[i] === undefined || this[i] === null):
+          str += separator;
+          break;
+        case true:
+          str += this[i] + String(separator);
+          break;
+      }
```

Comment:
you can just use 2 if's instead of all this cases
```
      if (this[i] !== undefined && this[i] !== null) {
        result += `${this[i]}`;
      }

      if (i < this.length - 1) {
        result += separator;
      }
```

Reply:
Ok, I understood that. Thank you